pkgs/development/libraries/mpfr/default.nix

```diff
@@ -9,23 +9,34 @@
 # cannot use fetchpatch! All mutable patches (generated by GitHub or
 # cgit) that are needed here should be included directly in Nixpkgs as
 # files.
-
-stdenv.mkDerivation rec {
-  version = "4.2.0";
+let
+  getPatchCount = version:
+    let versionParts = lib.splitVersion version; in
+    if builtins.length versionParts == 4 then lib.toInt (lib.last versionParts) else 0;
+
+  getVersionNoPatches = version:
+    lib.concatStringsSep "." (lib.take 3 (lib.splitVersion version));
+in
+stdenv.mkDerivation (finalAttrs: {
+  version = "4.2.0-12";
   pname = "mpfr";
 
   src = fetchurl {
-    urls = [
-      "https://www.mpfr.org/${pname}-${version}/${pname}-${version}.tar.xz"
-      "mirror://gnu/mpfr/${pname}-${version}.tar.xz"
+    urls = let name = "${finalAttrs.pname}-${getVersionNoPatches finalAttrs.version}"; in [
+      "https://www.mpfr.org/${name}/${name}.tar.xz"
+      "mirror://gnu/mpfr/${name}.tar.xz"
     ];
     hash = "sha256-BqN43xNQEkjBsttaqXeiyBJq6Emp2be+JUb7Spwm2ZM=";
   };
 
+  patches = builtins.genList
+    (i: builtins.toPath "${./patches}/patch${if i + 1 < 10 then "0" else ""}${toString (i + 1)}.patch")
```

Comment:
The magic number "10" here is error-prone; people will forget to update it.

Could you please have the `./update.sh` script `echo $patch_count > num-patches.nix` and then `./import num-patches.nix`?

Or better yet, just write the actual `[ ./1.patch ./2.patch ./3.patch ]` into a `patches.nix` and import that.